scripts/do-release.sh

```diff
@@ -21,12 +21,14 @@ function create_release() {
     local project="$1"
     local target="$2"
     local files=( "${@:3}" )
-    [[ "${release['pre-release']}" = "true" ]] && local prerelease="--prerelease"
-#    [[ -n "${release['release-notes']}" ]] && local notes="--notes ${release['release-notes']}"
+    local prerelease=false
+
+    [[ -z "${semver[pre]}" ]] || prerelease=true
 
     gh config set prompt disabled
     # shellcheck disable=SC2086,SC2068 # Some things have to be expanded or GH CLI flips out
-    with_retries 3 dryrun gh release create "${release['version']}" ${files[@]} ${prerelease} \
+    with_retries 3 dryrun gh release create "${release['version']}" ${files[@]} \
```

Comment:
```suggestion
    with_retries 3 dryrun gh release create "${release['version']}" "${files[@]}" \
```

Reply:
First I didn't change that part of the line.

Second, you entirely missed the comment in the line just above it..
`    # shellcheck disable=SC2086,SC2068 # Some things have to be expanded or GH CLI flips out`

Reply:
Yes, I know you didn’t change that part of the line, I just noticed this as part of the review. The way I understood it, the quote-less expansion is needed for `${prerelease}`. If it is needed for the `files` array (which I don’t think it is), the expansion should be `${files[*]}`; `@` expansions should always be quoted.

Reply:
Yes it was probably due to the prerelease flag, I tested on my fork and quoted files seems to work so I'll include this change.